design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,219 @@
+```
+
+or
+
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=default,kube-system
+    --include-cluster-resources=true
+```
+
+#### all namespaced resources + no additional cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=*
+```
+
+#### all namespaced resources + some additional cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=*
+    --include-cluster-scope-resources=PersistentVolume
+```
+
+#### all namespaced resources + all cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=*
+    --include-cluster-scope-resources=*
+```
+
+or
+
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=*
+    --include-cluster-resources=true
+```
+
+#### describe command change
+In `velero backup describe` command, the two new parameters should be outputted too.
+``` bash
+ velero backup describe <backup-name>
+......
+
+Namespaces:
+  Included:  ns2
+  Excluded:  <none>
+
+Resources:
+  Included:        *
+  Excluded:        <none>
+  Included-cluster-scope: StorageClass,PersistentVolume
+  Excluded-cluster-scope: <none>
+  Cluster-scoped:  auto
+
+......
+```
+
+**Note:** Similar function is implemented to `velero restore` commands too.
+
+## Detailed Design
+With adding `IncludedClusterScopeResources` and `ExcludedClusterScopeResources`, the `BackupSpec` and `RestoreSpec` look like:
+``` go
+type BackupSpec struct {
+    ......
+	// IncludedResources is a slice of resource names to include
+	// in the backup. If empty, all resources are included.
+	// Will be changed to only take effect on namespace scope resources in v1.11.
+	// Overwritten by IncludedClusterScopeResources in conflict case in v1.10.
+	// +optional
+	// +nullable
+	IncludedResources []string `json:"includedResources,omitempty"`
+
+	// ExcludedResources is a slice of resource names that are not
+	// included in the backup.
+	// Will be changed to only take effect on namespace scope resources in v1.11.
+	// Overwritten by ExcludedClusterScopeResources in conflict case in v1.10.
+	// +optional
+	// +nullable
+	ExcludedResources []string `json:"excludedResources,omitempty"`
+
+	// IncludeClusterResources specifies whether cluster-scoped resources
+	// should be included for consideration in the backup.
+	// Will be deprecated in v1.11. Replaced by `IncludedClusterScopeResources: *` for setting true.
+	// +optional
+	// +nullable
+	IncludeClusterResources *bool `json:"includeClusterResources,omitempty"`
+
+	// IncludedClusterScopeResources is a slice of cluster scope 
+	// resource names to include in the backup.
+	// If empty, no additional cluster scope resources are included.
+	// If set to "*", all cluster scope resources are included.
+	// Supersede IncludedResources in conflict case.
+	// +optional
+	// +nullable
+	IncludedClusterScopeResources []string `json:"includedClusterScopeResources,omitempty"`
```

Comment:
As mentioned on the call, lets also create `IncludedNamespacedResources` and `ExcludedNamespaceResources` where empty/"*" rules match the rules for the Cluster resources, and we want validation to ensure that a user either uses IncludedResources/ExcludedResources *or* some combination of the four new fields, but not both. Old fields work just like they do now. New fields respect the cluster/namespaced split.

Reply:
Added.